### PR TITLE
Drop support for autotools / automake / make

### PR DESCRIPTION
**Pull request by keszybz**
[NOT BEFORE v234 IS RELEASED]

I think the meson build system works quite nicely, and keeping two build systems is a burden. Let's drop autotools support some time after the v234 release.

Further cleanups are possible, for example to sanitize some `#define` names, and to switch to `-Werror=undef` (c.f. https://github.com/systemd/casync/commit/d40be70bdbef202c1d98f4d99422cfd007c41641), but that can be done later.

### CODE REVIEW

**Comment by poettering** [comment]
can't wait for this!

**Comment by ronnychevalier** [comment]
You also need to use `$BUILD_DIR` in `run_nspawn` instead of `../../`. Otherwise, it will not find systemd-nspawn unless `$BUILD_DIR == ../../`.

**Comment by keszybz** [comment]
Yup. Updated.

**Comment by keszybz** [comment]
I'll rebase this ~1 week after v234 and any post-release fixes have gone in.

**Comment by ronnychevalier** [comment]
[HACKING](https://github.com/systemd/systemd/blob/master/HACKING) is still mentioning `autogen.sh` and `make`. It should be updated.

**Comment by keszybz** [comment]
Updated:
- rebase
- HACKING is fixed

File: README [hunk 1]
```diff
@@ -145,7 +142,13 @@ REQUIREMENTS:
...
         libidn2 or libidn (optional)
         elfutils >= 158 (optional)
-        make, gcc, and similar tools
+        pkg-config
+        gperf >= 3.1
+        docbook-xsl (optional, required for documentation)
+        xsltproc    (optional, required for documentation)
+        python-lxml (optional, required to build the indices)
+        python, meson, ninja
+        gcc, awk, sed, and similar tools
```
Comment (mbiebl) [inline README]: Should m4 be added here?
With autotools this was sort of implied, but with meson it's probably less common to require m4.

**Comment by mbiebl** [comment]
Should we add wrappers for ./configure && make to support https://github.com/cgwalters/build-api ?

**Comment by mbiebl** [comment]
See https://github.com/ebassi/graphene/blob/master/configure and the related discussion at https://mail.gnome.org/archives/desktop-devel-list/2017-April/msg00080.html

**Comment by keszybz** [comment]
Updated:
- m4, grep are mentioned
- build-api

**Comment by poettering** [comment]
hmm, needs rebase, but I am all for this to go in now

**Comment by keszybz** [comment]
`× default` — systemd-centos-ci/slave/bootstrap.sh: line 42: ./autogen.sh: No such file or directory
CI will need to be adjusted.

**Comment by keszybz** [comment]
(BTW., updated.)

**Comment by keszybz** [comment]
Rebased.

**Comment by poettering** [comment]
Sooo, should we merge this now? looks all good to me. I figure the semaphore CI will explode immediately though, as it uses make to build... Maybe we should switch all CIs over first?

**Comment by keszybz** [comment]
I added `[ -x autogen.sh ] || exit` to semaphoreci config. @martinpitt updated the ubuntu Ci a while back. Only centos remains, but I think @zonque will have to look into that. I'll rebase again, and I think we're good to go.

**Comment by keszybz** [comment]
It seems my hack to disable autoconf build results in a hang. I just deleted that "thread" instead.

**Comment by poettering** [comment]
OK, then let's get this in now!

**Comment by evverx** [comment]
Does anybody know what happened to the `CentOS` CI?

**Comment by keszybz** [comment]
It was disabled by @zonque. It needed work, and nobody had spare time for that.

**Comment by evverx** [comment]
I hope this https://github.com/systemd/systemd-centos-ci/pull/15 helps a bit.

**Comment by evverx** [comment]
https://github.com/systemd/systemd-centos-ci/pull/15 has just been merged. It seems that the `CentOS` CI can be reenabled.

**Comment by martinpitt** [comment]
Will that just work? Last time I spoke to @zonque changes needed to be deployed by someone on the Red Hat side? But let's try, I re-enabled that test in GitHub.

**Comment by evverx** [comment]
In https://ci.centos.org/job/systemd-pr-build/3665/console I see the following:
```
>>> Executing remote command: 'systemd-centos-ci/slave/testsuite.sh' on n24.crusty.ci.centos.org
<<< Remote command finished after 0.1 seconds, return code = 255
```
It would be helpful to obtain the logs. But I'm not sure how to do that.

**Comment by evverx** [comment]
To sum up, the `CentOS` CI has been failing again since it was reenabled, so it was [disabled](https://github.com/systemd/systemd/pull/6552#issuecomment-320694438) again.

**Pull request merged**